### DISCUSSION TITLE
partialSafeParseDeep ?

### DISCUSSION THREAD

**Discussion by lukasa1993** [JacobWeisenburger/zod_utilz#6] [Q&A] [answered]
what is best way to use this with z.object with nested z.objects

**Comment by JacobWeisenburger** [comment]
Your question lacks specificity. Please add more info so I am able to help.

Otherwise, please feel free to refer to the docs:
https://github.com/JacobWeisenburger/zod_utilz#partialsafeparse

**Comment by lukasa1993** [comment]
@JacobWeisenburger heres example:

```TS
const a = z.object({
  a: z.string(),
  b: z.object({
    a: z.string(),
    b: z.string(),
  }),
});

const payload = {
  a: "1234",
  b: {
    b: "1",
  },
};

console.log(zu.partialSafeParse(a, payload).validData);

```

this returns 

```js
{ a: '1234' }
```

so with `partialDeepSafeParse`

maybe it can return

```js
{ a: '1234', b: { b: '1' } }
```

**Reply by JacobWeisenburger** [reply, marked as answer]
```ts
const a = z.object( {
    a: z.string(),
    b: z.object( {
        a: z.string(),
        b: z.string(),
    } ).partial(),
} )

const payload = {
    a: "1234",
    b: {
        b: "1",
    },
}

console.log( a.safeParse( payload ) )
// {
//   success: true,
//   data: {
//     a: "1234",
//     b: {
//       b: "1",
//     },
//   },
// }
```

**Reply by lukasa1993** [reply]
> ```ts
> const a = z.object( {
>     a: z.string(),
>     b: z.object( {
>         a: z.string(),
>         b: z.string(),
>     } ).partial(),
> } )
> 
> const payload = {
>     a: "1234",
>     b: {
>         b: "1",
>     },
> }
> 
> console.log( a.safeParse( payload ) )
> // {
> //   success: true,
> //   data: {
> //     a: "1234",
> //     b: {
> //       b: "1",
> //     },
> //   },
> // }
> ```

thanks but i can't edit all nested objects to partial i need to somehow manage from top level

**Reply by JacobWeisenburger** [reply]
that is a problem with `zod`, not `zod_utilz`. Sorry about that.

**Reply by lukasa1993** [reply]
its okay i wrote custom function since that extended one from this package seem to work
